lib/class-wp-rest-menu-items-batch-processor.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * Class to batch-process multiple menu items in a single request
+ *
+ * @see WP_REST_Posts_Controller
+ */
+class WP_REST_Menu_Items_Batch_Processor {
+
+	private $controller;
+
+	public function __construct( WP_REST_Menu_Items_Controller $controller ) {
+		global $wpdb;
+
+		$this->controller = $controller;
+		$this->wpdb = $wpdb;
+	}
+
+	public function process( $navigation_id, $tree ) {
+		$this->controller->ignore_position_collision = true;
+
+		$current_menu_items = wp_get_nav_menu_items( $navigation_id, array( 'post_status' => 'publish,draft' ) );
+
+		$result = $this->wpdb->query('START TRANSACTION');
```

Comment:
MySQL TRANSACTION are not supported by all hosts. This would require a change to WordPress's min requirements. Also the fact isn't fail gracefully here,  is not great.

Reply:
Agreed, I want to get it to a state where it uses MySQL transactions when they're available, and doesn't use them when they're not. I'm exploring:

1. Bulk validation before performing any sql operations 
2. Only performing SQL queries for menu items that are actually changed, not for all of them

Reply:
This is talking about a sizable change to WordPress core. I think this should really be discussed in a ticket on Trac.